src/clj/game/cards/hardware.clj

```diff
@@ -1678,6 +1678,31 @@
                                                (has-subtype? target "Icebreaker")))
                                 :type :recurring}}})
 
+(define-card "Simulchip"
+  {:constant-effects [{:type :card-ability-additional-cost
+                       :req (req (and (same-card? card target)
+                                      (let [pred #(and (runner? (first %))
+                                                       (installed? (first %))
+                                                       (program? (first %)))]
+                                        (zero? (+ (event-count state nil :runner-trash pred)
+                                                  (event-count state nil :corp-trash pred))))))
+                       :value [:program]}]
+   :abilities [{:async true
+                :label "Install a program from the heap"
+                :req (req (and (not (install-locked? state side))
```

Comment:
Love how clean Simulchip is now. Will be even cleaner once you implement the bit you spoke about, that the ability only fires if there are any valid `:choices`.

I'd think then you'd also move `install-locked?` into the `:choices` req, right?

Reply:
Uh, maybe? Will probably want to change how we handle `install-locked?` in that case.